### PR TITLE
mypy: Fixing implicit None errors

### PR DESCRIPTION
**Pull request by richardjgowers**
fixes issue #93

we fortran now

### CODE REVIEW

File: gufe/components/proteincomponent.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def to_openmm_positions(self) -> omm_unit.Quantity:
 
         return openmm_pos
 
-    def to_pdb_file(self, out_path: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase] = None) -> str:
+    def to_pdb_file(self, out_path: Optional[Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase]] = None) -> str:
```
Comment (richardjgowers) [inline gufe/components/proteincomponent.py:377]: I'm not sure these are correct as optional, from reading the function body I think we actually need something here as there isn't a default filename

File: gufe/network.py [hunk 2]
```diff
@@ -98,7 +98,9 @@ def _defaults(self):
 
     def to_graphml(self) -> str:
         """ """
+        raise NotImplementedError
```
Comment (richardjgowers) [inline gufe/network.py:101]: @dwhswenson I thought we had this already, or was this a different Network object?

File: gufe/components/proteincomponent.py (hunk 1 above)
Comment (dwhswenson) [inline gufe/components/proteincomponent.py:377]:
```suggestion
    def to_pdb_file(self, out_path: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase]) -> str:
```

Not optional, then make it required?

File: gufe/components/proteincomponent.py [hunk 3]
```diff
@@ -421,7 +421,7 @@ def to_pdb_file(self, out_path: Union[Union[str, bytes, PathLike[str], PathLike[
         return out_path
 
     def to_pdbx_file(
-        self, out_path: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase] = None
+        self, out_path: Optional[Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase]] = None
```
Comment (dwhswenson) [inline gufe/components/proteincomponent.py:424]:
```suggestion
        self, out_path: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], io.TextIOBase]
```

File: gufe/network.py (hunk 2 above)
Comment (dwhswenson) [reply gufe/network.py:101]: `LigandNetwork`, not `AlchemicalNetwork`. We actually don't use `AlchemicalNetwork` yet.

**Review by dwhswenson — CHANGES_REQUESTED** [review]
For things that aren't actually optional, just make them required?

Otherwise looks good.

**Pull request merged**